src/libServer/JSONConversion.cpp

```diff
@@ -196,42 +199,51 @@ bool JSONConversion::checkJsonTx(const Json::Value& _json) {
   if (ret) {
     if (!_json["nonce"].isIntegral()) {
       LOG_GENERAL(INFO, "Fault in nonce");
-      return false;
+      throw jsonrpc::JsonRpcException(Server::RPC_INVALID_PARAMETER,
+                                      "Nonce is not integral");
     }
     if (_json["amount"].isString()) {
       try {
         uint128_t amount(_json["amount"].asString());
       } catch (exception& e) {
         LOG_GENERAL(INFO, "Fault in amount " << e.what());
-        return false;
+        throw jsonrpc::JsonRpcException(Server::RPC_INVALID_PARAMETER,
+                                        "Amount invalid string");
       }
     } else {
       LOG_GENERAL(INFO, "Amount not string");
-      return false;
+      throw jsonrpc::JsonRpcException(Server::RPC_INVALID_PARAMETER,
+                                      "Amount invalid string");
     }
     if (!_json["version"].isIntegral()) {
       LOG_GENERAL(INFO, "Fault in version");
-      return false;
+      throw jsonrpc::JsonRpcException(Server::RPC_INVALID_PARAMETER,
+                                      "Version not integral");
     }
     if (_json["pubKey"].asString().size() != PUB_KEY_SIZE * 2) {
       LOG_GENERAL(INFO,
                   "PubKey size wrong " << _json["pubKey"].asString().size());
-      return false;
+      throw jsonrpc::JsonRpcException(Server::RPC_INVALID_PARAMETER,
```

Comment:
RPC_INVALID_ADDRESS_OR_KEY?

Reply:
The suggestion from dev is that have same code if the issue is in the JSON

Reply:
I see. Then it is fine.